### PR TITLE
[ui] Allow filtering the global graph by asset group

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

Fixes: https://github.com/dagster-io/dagster/issues/15355

This PR enables filtering the global asset graph by asset group, and brings asset group multi-selection to the global asset catalog.

image

I considered porting the fancy RunsFilter UI over to the asset catalog, but at the moment group is the only available filter, and the RunsFilter UI exposes a "Filter" button and a separate list of "filter tags", which we don't have a great place to show in the asset graph.  I think that having a single component with typeahead filtering and multi-selection may be best until we redesign this UI a bit more significantly.

## How I Tested These Changes

I tested this in both the global asset graph and the asset catalog in a repo that contains many groups, some of which have overlapping names. I also verified that clicking around the global graph does not clear your filter, and that reloading the pages preserves / rehydrates the filter state correctly.

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import * as React from 'react';
+
+import {AssetGroupSelector} from '../graphql/types';
+import {ClearButton} from '../ui/ClearButton';
+import {buildRepoPathForHuman} from '../workspace/buildRepoAddress';
+
+type Asset = {
+  definition: {
+    groupName: string | null;
+    repository: {name: string; location: {name: string}};
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx:13]: Use `RepoAddress`?

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}> = ({assetGroups, value, onChange}) => {
+  const repoContextNeeded = React.useMemo(() => {
+    // This is a bit tricky - the first time we find a groupName it sets the key to `false`.
+    // The second time, it sets the value to `true` + tells use we need to show the repo name
+    const result: {[groupName: string]: boolean} = {};
+    assetGroups.forEach(
+      (group) => (result[group.groupName] = result.hasOwnProperty(group.groupName)),
+    );
+    return result;
+  }, [assetGroups]);
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx:51]: I've got a similar UI for a Cloud feature, and came to the conclusion that we may as well just always show the code location name, unless perhaps there's only one code location.

**Review by hellendag — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx:13]: Ahh good call will fix!

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx (hunk 2 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx:51]: I like that -- I'll make this show the repo names whenever more than one repo is present. Right now it's cool that it only shows the context when necessary, but the size of the suggestions varies and it's a bit odd. Good call!

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetGroupSuggest.tsx:13]: Bah, it turns out this is a partial asset coming from the backend, so we'd need to add a translation layer to make it use RepoAddress here.

**Pull request merged**